### DISCUSSION TITLE
How to build the Hoppscotch desktop as https://tauri.localhost

### DISCUSSION THREAD

**Discussion by Leon-Luu** [hoppscotch/hoppscotch#4509] [General]
Hi everyone

I am trying to build the Hoppscotch selfhost-desktop. The default output from tauri is serving as  http (http://tauri.localhost).
How can we configure and build as https (https://tauri.localhost). Anyone can share the successful configuration example with this?
Please share the tauri.conf.json, vite.config.ts and any other config files and steps needed.

Thanks!

**Comment by liudonghua123** [comment]
For tauri v2, maybe you can configure this via [webviewurl](https://tauri.app/reference/config/#webviewurl).

Here is a simple example of `tauri.config.json`.

```json
{
  "tauri": {
    "build": {
      "devPath": "http://localhost:3000", // For development
      "distDir": "../dist" // For production
    },
    "serve": {
      "webviewUrl": "https://tauri.localhost" // The URL to load in the webview
    }
  }
}

```

**Comment by Leon-Luu** [comment]
hi @liudonghua123 thanks for suggestion. I tested to add the webviewUrl to the tauri.config.json but it is still not working. The application still serves as http://tauri.localhost. I also tested to add following to the  tauri.config.json but it is same impact..

    "security": {
      "devServer": {
        "scheme": "https",
        "hostname": "tauri.localhost",
        "port": 3000
      },
      "certPath": "../../certs/localhost-cert.pem",
      "keyPath": "../../certs/localhost-key.pem"
    },
